engine/src/main/java/org/terasology/config/flexible/SettingImpl.java

```diff
@@ -209,4 +214,19 @@ public String getHumanReadableName() {
     public String getDescription() {
         return description;
     }
+
+    @Override
+    public void setValueFromString(String valueString) {
+        if (value instanceof Integer) {
+            value = (T)(Integer)Integer.parseInt(valueString);
+        } else if (value instanceof Float) {
+            value = (T)(Float)Float.parseFloat(valueString);
+        } else if (value instanceof Double) {
+            value = (T)(Double)Double.parseDouble(valueString);
+        } else if (value instanceof String) {
+            value = (T)valueString;
+        } else {
```

Comment:
I don't like this block for three reasons:
1. It violates OCP because every new type needs a code change
2. The double cast looks odd
3. It supports only int, float, double and string but this is only visible in the code

I don't have a better alternative right now but I am sure we will find one :)

Reply:
Yeah, I don't like this part of the code either - one of the first that I plan on reworking, as soon as i get a bit of free time.

Reply:
I agree with @oniatus assessment. 

I now have the even stronger feeling we might want to go back to a Setting class that is abstract, -not- generic-based, and is the base for classes such as FloatSetting, IntSetting etc.

I guess one alternative would be delegate the parsing to type-specific parsing objects which can be registered somewhere at runtime and retrieved/used in this method.

Any other possibility?

Reply:
"delegate the parsing to type-specific parsing objects which can be registered somewhere at runtime"
Sounds a lot like gson adapters - can't say I'm the biggest fan of that.

I've implemented a hybrid generic-and-concerete solution which I feel is a decent compromise. Have a look at it.